src/unix/tcp.c

```diff
@@ -445,7 +447,14 @@ int uv__tcp_listen(uv_tcp_t* tcp, int backlog, uv_connection_cb cb) {
 
   /* Start listening for connections. */
   tcp->io_watcher.cb = uv__server_io;
-  uv__io_start(tcp->loop, &tcp->io_watcher, POLLIN);
+  err = uv__io_start(tcp->loop, &tcp->io_watcher, POLLIN);
+  if (err) {
+    /* clean things up */
+    uv__queue_remove(&tcp->handle_queue);
```

Comment:
This doesn't look correct to me. Why did you pick this approach / how did you test this?

Reply:
When the tcp_init fails it does that.

https://github.com/libuv/libuv/blob/dccacd05aa5d20f5c48013d49c7fb0d8ad9c7e24/src/unix/tcp.c#L132

Reply:
Yes but here we are past init, so uv_close would be necessary and will take care of that, won't it?

Reply:
That is so true. I just pushed a new revision.